AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -668,4 +673,115 @@ - (void)authTokenContext:(MSAuthTokenContext *)__unused authTokenContext didUpda
+                                                                                                   operation:nil
+                                                                                            deviceTimeToLive:writeOptions.deviceTimeToLive];
+
+                                                    } else {
+                                                      [self.dataOperationProxy.documentStore deleteWithToken:tokenResponses.tokens[0]
+                                                                                                  documentId:operation.documentId];
+                                                    }
+
+                                                  } else if (documentWrapper.error.errorCode == MSACDocumentNotFoundErrorCode ||
+                                                             documentWrapper.error.errorCode == MSACDocumentConflictErrorCode) {
+                                                    MSLogError(
+                                                        [MSDataStore logTag],
+                                                        @"Failed to call Cosmos delete API. Remote operation failed with error code: %ld",
+                                                        documentWrapper.error.errorCode);
+                                                    [self.dataOperationProxy.documentStore deleteWithToken:tokenResponses.tokens[0]
+                                                                                                documentId:operation.documentId];
+                                                  } else {
+                                                    MSLogError([MSDataStore logTag], @"Failed to call Cosmos create or replace API: %@",
+                                                               [documentWrapper.error.error localizedDescription]);
+                                                  }
+                                                }];
+                                       } else if ([operation.operation isEqualToString:kMSPendingOperationDelete]) {
+
+                                         // Perform delete operation
+                                         [MSDataStore
+                                             deleteWithPartition:(NSString *)operation.partition
+                                                      documentId:(NSString *)operation.documentId
+                                                    writeOptions:(MSWriteOptions * _Nullable) writeOptions
+                                               completionHandler:(MSDocumentWrapperCompletionHandler) ^(MSDocumentWrapper *
+                                                                                                         documentWrapper) {
+                                                 if (!documentWrapper.error) {
+                                                   [self.dataOperationProxy.documentStore deleteWithToken:tokenResponses.tokens[0]
```

Comment:
I don't understand this line - when we perform a DELETE and get back a cosmos Db Error, why must we delete the pending operation from the local cache?

My understanding is (implemented in Android SDK at https://github.com/Microsoft/AppCenter-SDK-Android/blob/94b91be7bed62254046bb5f28b4bf7b5e5f85afe/sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java#L939-L943):
```
when we get back a cosmos Db Error
if (error code is 404 || error code is 409 || operation already expired) 
	delete the operation from local cache
else
	do nothing
```